### PR TITLE
CActiveDesktop

### PR DESCRIPTION
**Pull request by learn-more**
JIRA issue: [CORE-15424](https://jira.reactos.org/browse/CORE-15424)

### CODE REVIEW

File: dll/win32/shell32/CActiveDesktop.h [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+BEGIN_COM_MAP(CActiveDesktop)
+    COM_INTERFACE_ENTRY_IID(IID_IActiveDesktop, IActiveDesktop)
+    //COM_INTERFACE_ENTRY_IID(IID_IActiveDesktopP, IActiveDesktopP)
+    //COM_INTERFACE_ENTRY_IID(IID_IADesktopP2, IADesktopP2)
+    COM_INTERFACE_ENTRY_IID(IID_IPropertyBag, IPropertyBag)
+END_COM_MAP()
+};
+
+
+#endif //
```
Comment (HBelusca) [inline dll/win32/shell32/CActiveDesktop.h]: "//" ??

**Comment by Biswa96** [comment]
I identified some of these functions with IDA in _Win2K3_ (I'm aware of these in Win10 reversing).  Is there any method I can discuss those with you developers, except IRC?

File: dll/win32/shell32/CActiveDesktop.cpp [hunk 2]
```diff
@@ -0,0 +1,156 @@
+/*
+ * PROJECT:     shell32
+ * LICENSE:     LGPL-2.1+ (https://spdx.org/licenses/LGPL-2.1+)
```
[the remark this answers is not in the document]
Comment (HBelusca) [reply dll/win32/shell32/CActiveDesktop.cpp]: too long to type I suppose? xD

File: dll/win32/shell32/CActiveDesktop.h [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    public CComCoClass<CActiveDesktop, &CLSID_ActiveDesktop>,
+    public CComObjectRootEx<CComMultiThreadModelNoCS>,
+    public IActiveDesktop,
+    //public IActiveDesktopP,
+    //public IADesktopP2,
+    public IPropertyBag
+{
+public:
+    CActiveDesktop();
+    ~CActiveDesktop();
```
[the remark this answers is not in the document]
Comment (learn-more) [reply dll/win32/shell32/CActiveDesktop.h]: Good point.

File: dll/win32/shell32/CActiveDesktop.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (learn-more) [reply dll/win32/shell32/CActiveDesktop.cpp]: Copypasta from another class

File: dll/win32/shell32/CMakeLists.txt [hunk 4]
```diff
@@ -25,6 +25,8 @@ include_directories(
     ${REACTOS_SOURCE_DIR})
 
 list(APPEND SOURCE
+    CActiveDesktop.cpp
+    CActiveDesktop.h
```
[the remark this answers is not in the document]
Comment (learn-more) [reply dll/win32/shell32/CMakeLists.txt]: It makes the file available in visual studio ;)

**Pull request merged**